### PR TITLE
pythonPackages.gast_0_2_2: init for tensorflow

### PR DESCRIPTION
**Pull request by matthewbauer**
It’s been reported that python-gast-0.2.2 is needed for Tensorflow
Autograph. See https://github.com/tensorflow/tensorflow/issues/38734.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by jonringer** [comment]
The link you provided just shows an install error. Is there a runtime error with using gast>=0.3?

Generally we try to avoid different versions in `python-modules` as this will create incoherent package sets if a user creates an envrionment with 2 different versions of the same package.

**Comment by matthewbauer** [comment]
A better description of the issue is here:

https://github.com/tensorflow/tensorflow/issues/32319

File: pkgs/development/python-modules/gast/0_2_2.nix [hunk 1]
```diff
@@ -0,0 +1,16 @@
+{ stdenv, fetchPypi, buildPythonPackage, astunparse }:
+
+buildPythonPackage rec {
+  pname = "gast";
+  version =  "0.2.2";
```
Comment (jonringer) [inline pkgs/development/python-modules/gast/0_2_2.nix:5]:
```suggestion
  version =  "0.2.2";

```

File: pkgs/development/python-modules/gast/0_2_2.nix [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "1w5dzdb3gpcfmd2s0b93d8gff40a1s41rv31458z14inb3s9v4zy";
+  };
```
Comment (jonringer) [inline pkgs/development/python-modules/gast/0_2_2.nix:9]:
```suggestion
  };

```

File: pkgs/development/python-modules/gast/0_2_2.nix [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  checkInputs = [ astunparse ] ;
```
Comment (jonringer) [inline pkgs/development/python-modules/gast/0_2_2.nix:10]:
```suggestion
  checkInputs = [ astunparse ] ;

```

**Review by jonringer — COMMENTED** [review]
cc @FRidh

**Review by FRidh — CHANGES_REQUESTED** [review]
No this is not possible. We cannot have multiple versions of packages because this will cause runtime conflicts when mixing packages relying on the different versions.

**Comment by FRidh** [comment]
Simply put, the older tensorflow will no longer work with Nixpkgs, unless the user themselves performs an override to fix it.

**Pull request closed without merging**